cmd/helm-sync/watch.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apiwatch "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/klog/v2"
+	api "kpt.dev/configsync/pkg/api/configsync/v1beta1"
+	"kpt.dev/configsync/pkg/client/restconfig"
+	"kpt.dev/configsync/pkg/remediator/watch"
+	"kpt.dev/configsync/pkg/util"
+	utillog "kpt.dev/configsync/pkg/util/log"
+)
+
+func watchCM(valuesFrom []*api.ValuesFrom, cmValuesMap map[string]string, refreshCM *bool, log *utillog.Logger) {
```

Comment:
This function is problematic, IMO. I don't like that we're invoking this function as a goroutine and communicating using `os.Exit` and pointers with a shared mutex.

Can you rewrite this to communicate with the main goroutine using channels?